pallets/token-manager/src/lib.rs

```diff
@@ -1023,21 +1034,35 @@ impl<T: Config> Pallet<T> {
         let event_validity = T::ProcessedEventsChecker::processed_event_exists(event_id);
         ensure!(event_validity, Error::<T>::NoTier1EventForLogLowerClaimed);
 
+        Self::remove_used_lower(data.lower_id)?;
+        Self::deposit_event(Event::<T>::AvtLowerClaimed { lower_id: data.lower_id });
+
+        Ok(())
+    }
+
+    fn remove_used_lower(lower_id: LowerId) -> DispatchResult {
         ensure!(
-            LowersReadyToClaim::<T>::contains_key(data.lower_id) == true,
+            LowersReadyToClaim::<T>::contains_key(lower_id) == true,
             Error::<T>::InvalidLowerId
         );
-        LowersReadyToClaim::<T>::remove(data.lower_id);
+        LowersReadyToClaim::<T>::remove(lower_id);
         ensure!(
-            LowersReadyToClaim::<T>::contains_key(data.lower_id) == false,
+            LowersReadyToClaim::<T>::contains_key(lower_id) == false,
             Error::<T>::InvalidLowerId
         );
 
-        Self::deposit_event(Event::<T>::AvtLowerClaimed { lower_id: data.lower_id });
-
         Ok(())
     }
 
+    fn process_lower_reverted(event: &EthEvent, data: &LowerRevertedData) -> DispatchResult {
+        let event_id = &event.event_id;
+        let event_validity = T::ProcessedEventsChecker::processed_event_exists(event_id);
+        ensure!(event_validity, Error::<T>::NoTier1EventForLogLowerReverted);
+
+        Self::remove_used_lower(data.lower_id)?;
+        Self::lift_to_account(event, data.token_contract, &data.t2_public_key, data.amount)
```

Comment:
I know we have a lift event but do we want to emit a specific `revert` event? It would help Dapps and the explorer track reverts.